bench/Main.hs

```diff
@@ -24,49 +27,63 @@ main = do
         runStateGen (mkStdGen 2020) $ \g -> replicateM 5000 (uniformRM (16 + 1, 16 + 7) g)
   defaultMain
     [ bgroup "baseline"
-      [ let !smGen = SM.mkSMGen 1337 in bench "nextWord32" $ nf (genMany SM.nextWord32 smGen) sz
-      , let !smGen = SM.mkSMGen 1337 in bench "nextWord64" $ nf (genMany SM.nextWord64 smGen) sz
-      , let !smGen = SM.mkSMGen 1337 in bench "nextInt" $ nf (genMany SM.nextInt smGen) sz
-      , let !smGen = SM.mkSMGen 1337 in bench "split" $ nf (genMany SM.splitSMGen smGen) sz
+      [ env (pure $ SM.mkSMGen $ fromIntegral seed) $ \smGen ->
+          bench "nextWord32" $ whnf (genMany SM.nextWord32 smGen) sz
+      , env (pure $ SM.mkSMGen $ fromIntegral seed) $ \smGen ->
+          bench "nextWord64" $ whnf (genMany SM.nextWord64 smGen) sz
+      , env (pure $ SM.mkSMGen $ fromIntegral seed) $ \smGen ->
+          bench "nextInt" $ whnf (genMany SM.nextInt smGen) sz
+      , env (pure $ SM.mkSMGen $ fromIntegral seed) $ \smGen ->
+          bench "split" $ whnf (genMany SM.splitSMGen smGen) sz
       ]
     , bgroup "pure"
       [ bgroup "random"
-        [ pureRandomBench (Proxy :: Proxy Float) sz
-        , pureRandomBench (Proxy :: Proxy Double) sz
-        , pureRandomBench (Proxy :: Proxy Integer) sz
+        [ env (pure (mkStdGen seed)) $ \gen ->
+            bench "Word32'" $ whnf (genMany (random :: StdGen -> (Word32, StdGen)) gen) sz
```

Comment:
Why do we need this additional benchmark?

Reply:
We don't. I just forgot to get rid of it.